lib/bokken_web/controllers/guardian_controller.ex

```diff
@@ -0,0 +1,47 @@
+defmodule BokkenWeb.GuardianController do
+  use BokkenWeb, :controller
+
+  alias Bokken.Accounts
+  alias Bokken.Accounts.Guardian
+
+  action_fallback BokkenWeb.FallbackController
+
+  def index(conn, _params) do
+    guardians = Accounts.list_guardians()
+    render(conn, "index.json", guardians: guardians)
+  end
+
+  def create(conn, %{"guardian" => guardian_params}) do
+    user = Accounts.create_user(guardian_params["user"])
+    user_id = elem(user, 1).id
+
+    with {:ok, %Guardian{} = guardian} <-
+           Accounts.create_guardian(Map.put(guardian_params, "user_id", user_id)) do
```

Comment:
What do you think of this requirement to create a user before you create the guardian to be included in the function `create_guardian`? That way we could make this a database transaction. I don't think it makes sense to create the a guardian without creating a user.

Reply:
I have been thinking, and because the flow should be first you sign up, and then you complete your registration, it's better two different functions like you did.